src/inmanta/agent/agent.py

```diff
@@ -724,18 +733,30 @@ async def repair_action() -> None:
                 self._deploy_splay_value,
                 (now + datetime.timedelta(seconds=self._deploy_splay_value)).strftime(const.TIME_LOGFMT),
             )
-            self._enable_time_trigger(deploy_action, self._deploy_interval, self._deploy_splay_value)
-        if self._repair_interval > 0:
+            interval_schedule_deploy: IntervalSchedule = IntervalSchedule(
+                interval=float(self._deploy_interval), initial_delay=float(self._deploy_splay_value)
+            )
+            self._enable_time_trigger(deploy_action, interval_schedule_deploy)
+        if isinstance(self._repair_interval, int):
```

Comment:
If we accept a dual type here I think we should make it clear in the type definition in the constructor.